scripts/validate_min_versions_in_sync.py

```diff
@@ -41,6 +51,127 @@
 import _optional
 
 
+def pin_min_versions_to_ci_deps():
+    all_yaml_files = list(YAML_PATH.iterdir())
+    all_yaml_files.append(ENV_PATH)
+    toml_dependencies = {}
+    with open(SETUP_PATH, "rb") as toml_f:
+        toml_dependencies = tomllib.load(toml_f)
+    for curr_file in all_yaml_files:
+        with open(curr_file) as yaml_f:
+            yaml_file_data = yaml_f.read()
+            yaml_file = yaml.safe_load(yaml_file_data)
+            yaml_dependencies = yaml_file["dependencies"]
+            res = []
+            [res.append(x) for x in yaml_dependencies if x not in res]
```

Comment:
is this to find the unique dependencies? perhaps using a set would be clearer, like
```python
res = list({x for x in yaml_dependencies})
```

Reply:
Yes. I was using a set before but when I would run pre-commit, I get this:
`TypeError: unhashable type: 'dict'`

Reply:
do you remember which hook was throwing that? pretty sure this is valid syntax
```python
In [15]: yaml_dependencies = [1,2,3, 3]

In [16]: list({x for x in yaml_dependencies})
Out[16]: [1, 2, 3]
```

Reply:
thanks, I'll take a look

Reply:
I think it is because of 
`tandard>=0.15.2', 'dask-core', 'seaborn-base', 'moto', 'flask', 'asv>=0.5.1', 'c-compiler', 'cxx-compiler', 'black=23.1.0', 'cpplint', 'flake8=6.0.0', 'isort>=5.2.1', 'mypy=1.0', 'pre-commit>=2.15.0', 'pyupgrade', 'ruff=0.0.215', 'gitpython', 'gitdb', 'natsort', 'numpydoc', 'pydata-sphinx-theme<0.11', 'pytest-cython', 'sphinx', 'sphinx-panels', 'sphinx-copybutton', 'types-python-dateutil', 'types-PyMySQL', 'types-pytz', 'types-setuptools', 'nbconvert>=6.4.5', 'nbsphinx', 'pandoc', 'ipywidgets', 'nbformat', 'notebook>=6.0.3', 'ipykernel', 'jinja2>=3.0.0', 'markdown', 'feedparser', 'pyyaml', 'requests', {'pip': ['sphinx-toggleprompt']}]`

In one of the yaml_dependencies, there is `{'pip': ['sphinx-toggleprompt']}]`

Reply:
ah, thanks for explaining!

Would it work to apply
```diff
$ git diff
diff --git a/scripts/validate_min_versions_in_sync.py b/scripts/validate_min_versions_in_sync.py
index b1dc734994..6e3c93dd57 100755
--- a/scripts/validate_min_versions_in_sync.py
+++ b/scripts/validate_min_versions_in_sync.py
@@ -62,9 +62,6 @@ def pin_min_versions_to_ci_deps():
             yaml_file_data = yaml_f.read()
             yaml_file = yaml.safe_load(yaml_file_data)
             yaml_dependencies = yaml_file["dependencies"]
-            res = []
-            [res.append(x) for x in yaml_dependencies if x not in res]
-            yaml_dependencies = res
             yaml_map = get_yaml_map_from(yaml_dependencies)
             toml_map = get_toml_map_from(toml_dependencies)
             yaml_file_data = pin_min_versions_to_yaml_file(
@@ -105,6 +102,8 @@ def get_yaml_map_from(yaml_dic) -> dict[str, str]:
     for dependency in yaml_dic:
         if type(dependency) == dict or dependency in EXCLUSION_LIST:
             continue
+        if dependency in yaml_map:
+            continue
         search_text = str(dependency)
         operator = get_operator_from(search_text)
         if "," in dependency:
```
?

I would help if you could also add a short little docstring to `pin_min_versions_to_ci_deps` in which you note that `pip` dependencies won't be pinned (which is definitely OK for now, what you've done already adds value)

Reply:
Nice, it works! I added the docstring, along with all the other requested updates.